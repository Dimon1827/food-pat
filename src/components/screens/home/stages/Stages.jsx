import advantages from './data.js';
import StageItem from './StageItem';
import './Stages.scss';

function Stages() {
  return (
    <section className="stages">
      <div className="container">
        <h2 className="stages__title title">Преимущества</h2>
        <ul className="stages__list">
          {!!advantages.length &&
            advantages.map(({ title, text }) => (
              <StageItem title={title} text={text} />
            ))}
        </ul>
      </div>
    </section>
  );
}

export default Stages;
