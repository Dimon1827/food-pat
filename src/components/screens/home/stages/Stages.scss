@import '@/assets/styles/variables';

.stages {
  padding: 23px 0 49px 0;
  background: linear-gradient(80deg, rgba(255, 255, 255, 1) 37%, rgba(255, 184, 65, 1) 87%);

  @media (min-width: $tablet-width) {
    padding: 45px 0 104px;
    background: linear-gradient(80deg, rgba(255, 255, 255, 1) 13%, rgba(255, 184, 65, 1) 87%);
  }

  @media (min-width: $desktop-width) {
    padding: 71px 0 128px;
  }

  .stages__title {
  margin-bottom: 42px;

  @media (min-width: $tablet-width) {
    margin-bottom: 60px;
  }
}
}

.stages__list {
  margin: 0;
  padding: 0;
  list-style-type: none;
  display: grid;
  gap: 30px;

  @media (min-width: $tablet-width) {
    grid-template-columns: repeat(2, 47%);
    gap: 40px;
  }

  @media (min-width: $desktop-width) {
    grid-template-columns: repeat(4, 1fr);
    gap: 69px 79px;
  }
}

.stages__item {
  text-align: center;
  position: relative;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.08);
  background: $white-color;
  padding: 22px 8px;
  padding-top: 142px;
  border-radius: 20px;

  &::before {
    position: absolute;
    content: "";
    top: 19px;
    left: 0;
    right: 0;
    margin-left: auto;
    margin-right: auto;
    width: 100px;
    height: 100px;
  }

  &:nth-child(1) {
    &::before {
      background: url("./icons/item-1.svg");
    }
  }

  &:nth-child(2) {
    &::before {
      background: url("./icons/item-2.svg");
    }
  }

  &:nth-child(3) {
    &::before {
      background: url("./icons/item-3.svg");
    }
  }

  &:nth-child(4) {
    &::before {
      background: url("./icons/item-4.svg");
    }
  }
}



.stages__subtitle {
  margin: 0;
  margin-bottom: 20px;
  font-weight: 600;
  font-size: 30px;
  line-height: 120%;
}

.stages__text {
  margin: 0;
  font-weight: 500;
  font-size: 20px;
  line-height: 120%;
  text-align: center;
  color: #666666;
}
