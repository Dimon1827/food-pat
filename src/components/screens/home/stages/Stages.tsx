import advantages from './data';
import StageItem from './StageItem';
import './Stages.scss';

function Stages() {
  return (
    <section className="stages">
      <div className="container">
        <h2 className="stages__title title">Преимущества</h2>
        <ul className="stages__list">
          {!!advantages.length &&
            advantages.map(({ title, text }, index) => (
              <StageItem key={index} title={title} text={text} />
            ))}
        </ul>
      </div>
    </section>
  );
}

export default Stages;
