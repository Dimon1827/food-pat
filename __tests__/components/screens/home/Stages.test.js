import { screen, render } from '@testing-library/react';
import Stages from '@/components/screens/home/stages/Stages';

describe('Stages', () => {
  it('Stages renders correctly', () => {
    render(<Stages />);
    const title = screen.getByText('Качество');
    const text = screen.getByText(
      'Вкусно, полезно, доступно - здоровое питание для всех'
    );
    expect(screen.getByTestId('stages')).toMatchSnapshot();
    expect(title).toBeInTheDocument();
    expect(text).toBeInTheDocument();
  });
});
