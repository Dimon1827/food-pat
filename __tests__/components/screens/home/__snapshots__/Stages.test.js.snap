// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stages Stages renders correctly 1`] = `
<section
  class="stages"
  data-testid="stages"
>
  <div
    class="container"
  >
    <h2
      class="stages__title title"
    >
      Преимущества
    </h2>
    <ul
      class="stages__list"
    >
      <li
        class="stages__item"
      >
        <h3
          class="stages__subtitle"
        >
          Качество
        </h3>
        <p
          class="stages__text"
        >
          Вкусно, полезно, доступно - здоровое питание для всех
        </p>
      </li>
      <li
        class="stages__item"
      >
        <h3
          class="stages__subtitle"
        >
          Многообразие
        </h3>
        <p
          class="stages__text"
        >
          Широкий выбор продуктов для активной и здоровой жизни
        </p>
      </li>
      <li
        class="stages__item"
      >
        <h3
          class="stages__subtitle"
        >
          Цена
        </h3>
        <p
          class="stages__text"
        >
          Качество по доступной цене! У нас выгодные цены на качественные продукты
        </p>
      </li>
      <li
        class="stages__item"
      >
        <h3
          class="stages__subtitle"
        >
          Подбор
        </h3>
        <p
          class="stages__text"
        >
          Поможем вам составить здоровый рацион и достичь целей
        </p>
      </li>
    </ul>
  </div>
</section>
`;
